src/tiled/tilesetview.cpp

```diff
@@ -620,6 +630,24 @@ void TilesetView::editTileProperties()
     mMapDocument->emitEditCurrentObject();
 }
 
+void TilesetView::swapTiles()
+{
+    if (selectionModel()->selectedIndexes().size() != 2)
+        return;
+
+    TileLayer *tileLayer = dynamic_cast<TileLayer*>(mMapDocument->currentLayer());
+    if (!tileLayer)
+        return;
+
+    const TilesetModel *model = tilesetModel();
+    Tile *tile1 = model->tileAt(selectionModel()->selectedIndexes()[0]);
+    Tile *tile2 = model->tileAt(selectionModel()->selectedIndexes()[1]);
```

Comment:
You're retrieving the list of selected indexes three times in this function. Please put it in a local variable.